clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -524,6 +524,58 @@ public void addMetric(MetricName metricName, MetricValueProvider<?> metricValueP
         addMetric(metricName, null, metricValueProvider);
     }
 
+    /**
+     * Create or get an existing metric to monitor an object that implements Measurable.
+     * This metric won't be associated with any sensor. This is a way to expose existing values as metrics.
+     *
+     * This method is kept for binary compatibility purposes, it has the same behaviour as
+     * {@link #metricOrElseCreate(MetricName, MetricConfig, MetricValueProvider)}.
+     *
+     * @param metricName The name of the metric
+     * @param config The configuration to use when measuring this measurable
+     * @param measurable The measurable that will be measured by this metric
+     * @return Existing KafkaMetric if already registered or else a newly created one
+     */
+    public KafkaMetric metricOrElseCreate(MetricName metricName, MetricConfig config, Measurable measurable) {
```

Comment:
I wonder if we really need all these overloads. Do we have use cases for all of them? I have the impression that the last one would be sufficient.

Reply:
I removed the other 2. I had added them to be at par with the addMetric method and it's overloads.